iotedgehubdev/utils.py

```diff
@@ -149,3 +146,24 @@ def exe_proc(params, shell=False, cwd=None, suppress_out=False):
             raise
         except Exception as e:
             raise Exception("Error while executing command: {0}. {1}".format(' '.join(params), str(e)))
+
+    @staticmethod
+    @suppress_all_exceptions()
+    def hash_connection_str_hostname(connection_str):
+        """Hash connection string hostname to count distint IoT Hub number"""
+        connection_str_dict = Utils.parse_device_connection_str(connection_str)
+        hostname = connection_str_dict[EC.HOSTNAME_KEY]
+
+        # get hostname suffix (e.g., azure-devices.net) to distinguish national clouds
+        if "." in hostname:
+            hostname_suffix = hostname[hostname.index(".") + 1:]
+        else:
+            hostname_suffix = ""
+
+        return (Utils.get_sha256_hash(hostname), hostname_suffix)
```

Comment:
What's the hashed value if hostname is empty string? We'd better return empty string in this case.

Reply:
Updated. Please help check again.